examples/math/sampling_primitives.rs

```diff
@@ -0,0 +1,624 @@
+    mut motion_events: EventReader<MouseMotion>,
+    mut scroll_events: EventReader<MouseWheel>,
+    mut camera: Query<&mut CameraRig>,
+    mut mouse_pressed: ResMut<MousePressed>,
+) {
+    // Store left-pressed state in the MousePressed resource
+    for button_event in button_events.read() {
+        if button_event.button != MouseButton::Left {
+            continue;
+        }
+        *mouse_pressed = MousePressed(button_event.state.is_pressed());
+    }
+
+    let mut camera_rig = camera.single_mut();
+
+    let mouse_scroll = scroll_events
+        .read()
+        .fold(0.0, |acc, scroll_event| acc + scroll_event.y);
+    camera_rig.distance -= mouse_scroll / 15.0 * MAX_CAMERA_DISTANCE;
+    camera_rig.distance = camera_rig
+        .distance
+        .clamp(MIN_CAMERA_DISTANCE, MAX_CAMERA_DISTANCE);
+
+    // If the mouse is not pressed, just ignore motion events
+    if !mouse_pressed.0 {
+        return;
+    }
+    let displacement = motion_events
+        .read()
+        .fold(Vec2::ZERO, |acc, mouse_motion| acc + mouse_motion.delta);
+    camera_rig.yaw += displacement.x / 90.;
+    camera_rig.pitch += displacement.y / 90.;
+    // The extra 0.01 is to disallow weird behaviour at the poles of the rotation
+    camera_rig.pitch = camera_rig.pitch.clamp(-PI / 2.01, PI / 2.01);
+}
+
+#[allow(clippy::too_many_arguments)]
+fn spawn_points(
+    mut commands: Commands,
+    mode: ResMut<Mode>,
+    shapes: Res<SampledShapes>,
+    mut random_source: ResMut<RandomSource>,
+    sample_mesh: Res<PointMesh>,
+    sample_material: Res<PointMaterial>,
+    mut spawn_queue: ResMut<SpawnQueue>,
+    mut counter: ResMut<PointCounter>,
+    spawn_mode: ResMut<SpawningMode>,
+) {
+    if let SpawningMode::Automatic = *spawn_mode {
+        spawn_queue.0 += POINTS_PER_FRAME;
+    }
+
+    if spawn_queue.0 == 0 {
+        return;
+    }
+
+    let rng = &mut random_source.0;
+
+    // Don't go crazy
+    for _ in 0..1000 {
+        if spawn_queue.0 == 0 {
+            break;
+        }
+        spawn_queue.0 -= 1;
+        counter.0 += 1;
+
+        let (shape, offset) = shapes.0.choose(rng).expect("There is at least one shape");
+
+        // Get a single random Vec3:
+        let sample: Vec3 = *offset
+            + match *mode {
+                Mode::Interior => shape.sample_interior(rng),
+                Mode::Boundary => shape.sample_boundary(rng),
+            };
+
+        // Spawn a sphere at the random location:
+        commands.spawn((
+            PbrBundle {
+                mesh: sample_mesh.0.clone(),
+                material: match *mode {
+                    Mode::Interior => sample_material.interior.clone(),
+                    Mode::Boundary => sample_material.boundary.clone(),
+                },
+                transform: Transform::from_translation(sample),
+                ..default()
+            },
+            SamplePoint,
+        ));
+    }
+}
+
+fn despawn_points(
```

Comment:
Visually, I think that fading these points out before despawn would work great.

Reply:
Agreed, fading in would also be really neat imo.
I didn't know the best way to do so and I didn't want to tack-on another hack so I left it out.
If anyone has an idea how to do it neatly I'll add it in.

Reply:
Mutate the light's intensity over time using a system that checks the delta-time. And then when the intensity reaches 0 despawn them.

Reply:
The issue is that the lights are not actually lights on the points, they're a combination of larger lights (which aggregate the "light" from all the points) and emissive colours on the Material (which is shared among all the points for performance and simplicity).

What could work is if the points were scaled to "pop in" or sizzle out, they'd get a special component for doing so.

I'll take a look on how difficult that would be, but it sounds reasonable.